Refactor namespaces

ahouseholder opened the pull request:
- Resolves #751 
- Resolves #754 
- Prepares the way for resolution of #592 in a future PR.

This PR refactors the python module layout so that namespaced objects are organized by namespace (pushing `ssvc` namespace objects into their own directory so they are peers to other namespaces like `cisa` etc.). This results in a lot of files being renamed and a lot of python import statement changes. Some other incidental changes are also included.

Sorry the PR is so large, it was not possible to do the refactoring without touching a lot of files.

## CoPilot Summary

This pull request includes updates to documentation and JSON files to improve clarity and consistency in terminology. The changes primarily involve replacing "no" with "negligible" in descriptions and updating testing instructions in the `README.md` file.

### Documentation Updates:
* [`README.md`](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L143): Updated testing instructions to clarify that tests run in the host OS, and removed references to Docker.

### Terminology Consistency in JSON Files:
* [`data/json/decision_points/cvss/modified_availability_impact_to_the_subsequent_system_1_0_0.json`](diffhunk://#diff-07e21bc70270599c66311d12399aef3c865ee0ce57ed2d8c28a59f786d84078bL12-R12): Replaced "no impact" with "negligible impact" in the description of availability impact.
* [`data/json/decision_points/cvss/modified_confidentiality_impact_to_the_subsequent_system_1_0_0.json`](diffhunk://#diff-84fa2df7a334980f48f229504e93eaff8d7978302b69f2968694e23756b86036L12-R12): Replaced "no loss of confidentiality" with "negligible loss of confidentiality" in the description of confidentiality impact.
* [`data/json/decision_points/cvss/modified_integrity_impact_to_the_subsequent_system_1_0_0.json`](diffhunk://#diff-f7c3396b9417fce867f853af15b97bd75e97731cd2a83a04d268b2d6f275eb46L12-R12): Replaced "no loss of integrity" with "negligible loss of integrity" in the description of integrity impact.

ahouseholder commented:
@sei-vsarvepalli I think I've fixed the problems you found.

- New makefile target `make regenerate_json` added so it's reproducible
- Fixed up `doctools.py` so it's walking directories for python modules

sei-vsarvepalli commented:
> @sei-vsarvepalli I think I've fixed the problems you found.
> 
> * New makefile target `make regenerate_json` added so it's reproducible
> * Fixed up `doctools.py` so it's walking directories for python modules

Still some issues.. The files should be generated under `data/json/outcomes` folder. 

```
data/json/decision_points/x_community/theparanoids_1_0_0.json
data/json/decision_points/x_basic/yes_no_1_0_0.json
data/json/decision_points/x_basic/moscow_1_0_0.json
data/json/decision_points/x_basic/value_complexity_1_0_0.json
data/json/decision_points/x_basic/do_schedule_delegate_delete_1_0_0.json
data/json/decision_points/cisa/cisa_levels_1_0_0.json
```




The file `src/ssvc/decision_points/ssvc/mission_prevalence.py` still has

`from ssvc.decision_points.ssvc_.base import SsvcDecisionPoint`

We can probably delete the whole mission_prevalence.py looks like it is not used anymore anywhere. The old Mission Prevalence may be gone now?

ahouseholder commented:
So the outcome groups are now decision point objects (which lets them be used as inputs to downstream decisions). Their appearance in the decision_point json dirs was intentional.

Mission Prevalence only ever showed up in CISA's model, so I just fixed it and moved it to the CISA namespace. I don't want to start deleting things that were used previously though, so I think it's okay to keep it for now.

sei-vsarvepalli reviewed (approved):
Looks good now.
